apm-agent-plugins/apm-jmx-plugin/src/main/java/co/elastic/apm/agent/jmx/JmxMetricTracker.java

```diff
@@ -69,20 +73,83 @@ public JmxMetricTracker(ElasticApmTracer tracer) {
         super(tracer);
         jmxConfiguration = tracer.getConfig(JmxConfiguration.class);
         metricRegistry = tracer.getMetricRegistry();
+    }
+
+    @Override
+    public void start(ElasticApmTracer tracer) {
+        ConfigurationOption.ChangeListener<List<JmxMetric>> initChangeListener = new ConfigurationOption.ChangeListener<>() {
+            @Override
+            public void onChange(ConfigurationOption<?> configurationOption, List<JmxMetric> oldValue, List<JmxMetric> newValue) {
+                if (oldValue.isEmpty() && !newValue.isEmpty()) {
+                    tryInit();
+                }
+            }
+        };
+        // adding change listener before checking if option is not empty to avoid missing an update due to a race condition
+        jmxConfiguration.getCaptureJmxMetrics().addChangeListener(initChangeListener);
+        if (!jmxConfiguration.getCaptureJmxMetrics().get().isEmpty()) {
+            tryInit();
+            jmxConfiguration.getCaptureJmxMetrics().removeChangeListener(initChangeListener);
+        } else {
+            logger.debug("Deferring initialization of JMX metric tracking until capture_jmx_metrics is set.");
+        }
+    }
 
+    private synchronized void tryInit() {
+        if (this.server != null || this.logManagerPropertyPoller != null) {
+            return;
+        }
         // Avoid creating the platform MBean server, only get it if already initialized
         // otherwise WildFly fails to start with a IllegalStateException:
         // WFLYLOG0078: The logging subsystem requires the log manager to be org.jboss.logmanager.LogManager
-        if (!MBeanServerFactory.findMBeanServer(null).isEmpty()) {
-            onPlatformMBeanServerInitialized(MBeanServerFactory.findMBeanServer(null).get(0));
+        if (setsCustomLogManager()) {
+            if (!MBeanServerFactory.findMBeanServer(null).isEmpty()) {
+                // platform MBean server is already initialized
+                init(MBeanServerFactory.findMBeanServer(null).get(0));
+            } else {
+                deferInit();
+            }
+        } else {
+            init(ManagementFactory.getPlatformMBeanServer());
         }
     }
 
-    @VisibleForAdvice
-    public void onPlatformMBeanServerInitialized(final MBeanServer platformMBeanServer) {
+    private void deferInit() {
+        logger.debug("Deferring initialization of JMX metric tracking until log manager is initialized");
+        Thread thread = new Thread(new Runnable() {
+
+            private final long timeout = System.currentTimeMillis() + TimeUnit.MINUTES.toMillis(5);
```

Comment:
I would set to longer. Sometimes servlet containers take longer to start up. May be rare, but the cost is not a problem, as the thread is already created.

Reply:
Does 10 min sound better? Note that it does not require all web application to fully be started. Logging initialization is one of the first things on startup so think that 5 min should be more than enough already.

Reply:
Sure, let's go with 5 and change if someone complains, just suggested because it doesn't have a real cost